### PR TITLE
Cfy 5395 create image

### PR DESCRIPTION
**Pull request by szpotona**
Add image creation

### CODE REVIEW

File: glance_plugin/image.py [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+@operation
+@with_glance_client
+def create(glance_client, **kwargs):
+    if use_external_resource(ctx, glance_client, IMAGE_OPENSTACK_TYPE):
+        return
+
+    image_dict = {
+        'name': get_resource_id(ctx, IMAGE_OPENSTACK_TYPE)
+    }
+    img_properties = ctx.node.properties['image']
```
Comment (Jessiebr) [inline glance_plugin/image.py]: use either img or image (for naming)

File: glance_plugin/tests/resources/test-image-start.yaml [hunk 2]
```diff
@@ -0,0 +1,29 @@
+
+tosca_definitions_version: cloudify_dsl_1_2
```
Comment (Jessiebr) [inline glance_plugin/tests/resources/test-image-start.yaml]: use cloudify_dsl_1_3

File: glance_plugin/tests/resources/test-image-start.yaml [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+imports:
+  - http://www.getcloudify.org/spec/cloudify/3.3.1/types.yaml
```
Comment (Jessiebr) [inline glance_plugin/tests/resources/test-image-start.yaml]: use 3.4 version- http://www.getcloudify.org/spec/cloudify/3.4/types.yaml

File: glance_plugin/tests/resources/test-image-start.yaml [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+  - plugin.yaml
```
Comment (Jessiebr) [inline glance_plugin/tests/resources/test-image-start.yaml]: import this instead- https://raw.githubusercontent.com/cloudify-cosmo/cloudify-openstack-plugin/master/plugin.yaml

File: glance_plugin/tests/resources/test-image-start.yaml [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+  use_password:
+    type: boolean
+    default: false
+
+node_templates:
+  image:
+    type: cloudify.openstack.nodes.Image
+    properties:
+      image:
+        url: test-url
```
Comment (Jessiebr) [inline glance_plugin/tests/resources/test-image-start.yaml]: need to be change to:

File: glance_plugin/tests/resources/test-image-start.yaml [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+      openstack_config:
+        username: aaa
+        password: aaa
+        tenant_name: aaa
+        auth_url: aaa
+        glance_url: aaa
```
Comment (Jessiebr) [inline glance_plugin/tests/resources/test-image-start.yaml]: no need glance_url

File: plugin.yaml [hunk 7]
```diff
@@ -466,6 +466,46 @@ node_types:
             openstack_config:
               default: {}
 
+  cloudify.openstack.nodes.Image:
+    derived_from: cloudify.nodes.Root
+    properties:
+      image:
+        default: {}
```
Comment (Jessiebr) [inline plugin.yaml]: will be good to provide description of fields and if required

File: glance_plugin/tests/resources/test-image-start.yaml [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+    type: boolean
+    default: false
+
+node_templates:
+  image:
+    type: cloudify.openstack.nodes.Image
+    properties:
+      image_url: test_url
+      image:
+        data_format: test_format
```
Comment (Jessiebr) [inline glance_plugin/tests/resources/test-image-start.yaml]: this should be disk_format

File: glance_plugin/tests/resources/test-image-start.yaml [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+inputs:
+  use_password:
+    type: boolean
+    default: false
+
+node_templates:
+  image:
+    type: cloudify.openstack.nodes.Image
+    properties:
+      image_url: test_url
```
Comment (Jessiebr) [inline glance_plugin/tests/resources/test-image-start.yaml]: provide image_url or path under image

Comment (szpotona) [inline glance_plugin/tests/resources/test-image-start.yaml]: Image_url should be outside - at least I couldn't find in glance docs an attribute to image.create() that would work with url. I'll remove image_url from this test.

**Pull request merged**